subprojects/logging/src/main/java/org/gradle/internal/logging/slf4j/DefaultContextAwareTaskLogger.java

```diff
@@ -0,0 +1,481 @@
+    public boolean isTraceEnabled(Marker marker) {
+        return delegate.isTraceEnabled(marker);
+    }
+
+    @Override
+    public boolean isDebugEnabled() {
+        return delegate.isDebugEnabled();
+    }
+
+    @Override
+    public boolean isDebugEnabled(Marker marker) {
+        return delegate.isDebugEnabled(marker);
+    }
+
+    @Override
+    public boolean isInfoEnabled() {
+        return delegate.isInfoEnabled();
+    }
+
+    @Override
+    public boolean isInfoEnabled(Marker marker) {
+        return delegate.isInfoEnabled(marker);
+    }
+
+    @Override
+    public boolean isWarnEnabled() {
+        return delegate.isWarnEnabled();
+    }
+
+    @Override
+    public boolean isWarnEnabled(Marker marker) {
+        return delegate.isWarnEnabled(marker);
+    }
+
+    @Override
+    public boolean isErrorEnabled() {
+        return delegate.isErrorEnabled();
+    }
+
+    @Override
+    public boolean isErrorEnabled(Marker marker) {
+        return delegate.isErrorEnabled(marker);
+    }
+
+    public void trace(String msg) {
+    }
+
+    public void trace(String format, Object arg) {
+    }
+
+    public void trace(String format, Object arg1, Object arg2) {
+    }
+
+    public void trace(String format, Object... arguments) {
+    }
+
+    public void trace(String msg, Throwable t) {
+    }
+
+    public void trace(Marker marker, String msg) {
+    }
+
+    public void trace(Marker marker, String format, Object arg) {
+    }
+
+    public void trace(Marker marker, String format, Object arg1, Object arg2) {
+    }
+
+    public void trace(Marker marker, String format, Object... argArray) {
+    }
+
+    public void trace(Marker marker, String msg, Throwable t) {
+    }
+
+    private void log(LogLevel logLevel, Throwable throwable, String message) {
+        OperationIdentifier buildOperationId = CurrentBuildOperationRef.instance().getId();
+        if (buildOperationId == null) {
+            buildOperationId = fallbackOperationIdentifier;
+        }
+        delegate.log(logLevel, throwable, message, buildOperationId);
+    }
+
+    private void log(LogLevel logLevel, Throwable throwable, String format, Object arg) {
+        log(logLevel, throwable, format, new Object[]{arg});
+    }
+
+    private void log(LogLevel logLevel, Throwable throwable, String format, Object arg1, Object arg2) {
+        log(logLevel, throwable, format, new Object[]{arg1, arg2});
+    }
+
+    private void log(LogLevel logLevel, Throwable throwable, String format, Object[] args) {
+        FormattingTuple tuple = MessageFormatter.arrayFormat(format, args);
+        Throwable loggedThrowable = throwable == null ? tuple.getThrowable() : throwable;
+
+        log(logLevel, loggedThrowable, tuple.getMessage());
+    }
+
+    public void debug(String message) {
+        if (isDebugEnabled()) {
+            log(LogLevel.DEBUG, null, message);
+        }
+    }
+
+    public void debug(String format, Object arg) {
+        if (isDebugEnabled()) {
```

Comment:
Could all of these not be done using a `delegate.isEnabled(LogLevel)` inside `log`? Seems like a bit less repetition.

Reply:
We would end up with multiple isEnabled check then instead of only do it once per log message as those log() methods are nested

Reply:
I was thinking in the last one that logs to the delegate, but I guess in some cases the message has already been formatted. OK, withdrawn.